duckduckgo/iOS
# [DuckPlayer] 20. OverlayInteracted

Task/Issue URL: https://app.asana.com/0/1204099484721401/1208057529481153/f

**Description**:
- Implement/Fix `overlayInteracted` mode.



**Steps to test this PR**:
1. Override the privacy config with https://jsonblob.com/api/1253721903368888320
2. Drag C.S.S scripts from [pr-releases/pr-993](https://github.com/duckduckgo/content-scope-scripts/tree/pr-releases/pr-993) to the project (Make sure it's up to date)
3. Go to DuckPlayer Setting and set it to "Ask Every Time"
4. Go to YouTube and Open a video (You should see the new overlay)
5. Enable the "Remember my choice" control and Tap on "No Thanks"
6. Navigate to another video on YouTube.
7. Confirm no Overlay is not shown anymore.
8. Go to Settings and confirm that DuckPlayer is STILL in "Ask Every Time" mode. (This is expected; it should be in this mode)
9. Select "Always" and then immediately select "Ask Every Time"
10. Go back to YouTube, and navigate to another video
11. Confirm The overlay is shown again.
